### DISCUSSION TITLE
IntelliJ: class rename not detected in user source directories

### DISCUSSION THREAD

**Discussion by juhalindfors** [jbangdev/jbang#1438] [Q&A] [answered]
If I create a Java class ```Something.java``` either as the main Java class or somewhere in the ```//SOURCES``` files and then rename the class in IntelliJ using refactoring tools to ```Foo.java```  then this rename is only reflected in the JBang's project cache directory but not in my actual source directories.

This is trivial to notice and fix in a single Java file projects but is easy to miss when working with multiple files using the ```//SOURCES``` directive. When prototyping new code, rename refactoring can be frequent.

Note that the source file's content is still updated correctly in the user source directory -- only the class filename fails to update. So you end up with a filename ```Something.java``` that contains ```class Foo```.

Is this a known issue or possibly a new issue?  

EDIT -- additional observations wrt file synchronization with JBang's project cache:

Loading and editing a script containing ```//SOURCES``` directive is picked up correctly in IntelliJ IDEA. All referenced source files are shown in the project tree under ```src``` directory. Again however, adding new Java classes is only reflected in JBang's project cache -- these new added files (classes) are not copied over to the user's home directory.

This quickly turns into a mess where the cache and user work directory are out-of-sync.

In general it appears that file create/delete/move operations are not correctly linked between the project cache and user directory.

**Comment by quintesse** [comment, marked as answer]
Just to confirm, but this is when you use `jbang edit` on a .java file, right?

**Reply by juhalindfors** [reply]
Yes that's correct.

**Reply by quintesse** [reply]
Then it's indeed something we're aware of, it's actually a long-standing issue where we have several ideas on how to handle this, but none of them fix the problem a 100%. These are some of the related issues/discussions: https://github.com/jbangdev/jbang/issues/462 and https://github.com/jbangdev/jbang/discussions/1282

**Reply by maxandersen** [reply]
Current way to work around this is to use the jbang plug-ins for vscode, eclipse or IntelliJ which avoids use of the temporary project which is what causes the problem.
